i18n/i18n.go

```diff
@@ -1,32 +1,105 @@
 package i18n
 
 import (
+	"encoding/json"
 	"fmt"
 	"path"
-	"sort"
+	"path/filepath"
 	"strings"
 
-	"github.com/nicksnyder/go-i18n/i18n"
-	"github.com/nicksnyder/go-i18n/i18n/language"
+	"github.com/nicksnyder/go-i18n/v2/i18n"
+	"golang.org/x/text/language"
 
 	"github.com/IBM-Cloud/ibm-cloud-cli-sdk/resources"
 )
 
 const (
 	defaultLocale   = "en_US"
-	resourcesSuffix = ".all.json"
+	resourcesSuffix = ".json"
+	resourcesPrefix = "all."
 )
 
-var T i18n.TranslateFunc
+var (
+	bundle        *i18n.Bundle
+	T             TranslateFunc
+	RESOURCE_PATH = filepath.Join("i18n", "resources")
+)
 
 func init() {
-	loadAsset("i18n/resources/" + defaultLocale + resourcesSuffix)
-	T = Tfunc(defaultLocale)
+	bundle = Bundle()
+	resource := resourcesPrefix + defaultLocale + resourcesSuffix
+	loadAsset(filepath.Join(RESOURCE_PATH, resource))
+	T = MustTfunc(defaultLocale)
+}
+
+// Bundle returns an instance of i18n.bundle
+func Bundle() *i18n.Bundle {
+	if bundle == nil {
+		bundle = i18n.NewBundle(language.AmericanEnglish)
+		bundle.RegisterUnmarshalFunc("json", json.Unmarshal)
+	}
+	return bundle
+}
+
+// Translate returns a method based on translate method signature in v1.3.0.
+// To allow compatability between v1.30 and v2.0+, the `messageId` and `args` parameters will
+// transformed to fit with the new Localize API method.
+// @see https://github.com/nicksnyder/go-i18n/blob/v1.3.0/i18n/bundle/bundle.go#L227-L257 for more
+// information on the translate method
+func Translate(loc *i18n.Localizer) TranslateFunc {
+	return func(messageId string, args ...interface{}) string {
+		var pluralCount interface{}
+		var templateData interface{}
+
+		/**
+		 * For the common usecases we can expect two scenarios. Below are two examples:
+		 *  1) T("login_error",  map[string]interface{}{"Command": "ibmcloud login"})
+		 *  2) T("login_fail_count", "2", map[string]interface{}{"Command": "ibmcloud login"})
+		 *
+		 * First paramter is always the `messageId`
+		 * Second paramter can be either pluralCount or templateData.
+		 * Third parameter can be templateData only if the second paramters is the plural count
+
+		 * If we have 2 args than we should expect scenario 2, otherwise we will assume scenario 1
+		 */
+		if argc := len(args); argc > 0 {
+			if isNumber(args[0]) {
+				pluralCount = args[0]
+				if argc > 1 {
+					templateData = args[1]
+				}
+			} else {
+				templateData = args[0]
+			}
+
+		}
+
+		msg, err := loc.Localize(&i18n.LocalizeConfig{
+			MessageID:    messageId,
+			TemplateData: templateData,
+			PluralCount:  pluralCount,
+		})
+
+		// If no message is returned we can assume that that
+		// the translation could not be found in any of the files
+		// panic and abort
+		if msg == "" {
+			panic(err)
```

Comment:
Would we have to panic here, or could we return something like "Message not found" ? I think the design of i18n is to panic, but might we deviate here to prevent a potentially ugly failure for customers?

Reply:
Yeah, good point